src/shuffle.h

```diff
@@ -27,7 +28,15 @@ class ShuffleChain {
     ShuffleChain() : ShuffleChain(1){};
 
     // Create a new ShuffleChain with the given window length.
-    explicit ShuffleChain(size_t window) : _max_window(window){};
+    explicit ShuffleChain(size_t window) : _max_window(window) {
+        std::random_device rd;
+        _rnd_generator.seed(rd());
+    }
+
+    // Create a new ShuffleChain with the given window length
+    // and using the given RandomNumberEngine
+    ShuffleChain(unsigned window, std::mt19937 rg)
```

Comment:
`window` should be size_t here as well to match.

Reply:
Damn, I overlooked this when rebasing after my sizes PR, will fix.